sql/src/test/java/io/crate/planner/SubQueryPlannerTest.java

```diff
@@ -33,31 +36,34 @@
 import org.junit.Test;
 
 import java.util.List;
+import java.util.function.Predicate;
 
 import static org.hamcrest.Matchers.instanceOf;
-
+import static org.hamcrest.core.Is.is;
 
 public class SubQueryPlannerTest extends CrateDummyClusterServiceUnitTest {
 
     private SQLExecutor e;
 
     @Before
     public void setUpExecutor() throws Exception {
-         e = SQLExecutor.builder(clusterService).addDocTable(T3.T1_INFO).build();
+        e = SQLExecutor.builder(clusterService).addDocTable(T3.T1_INFO).build();
     }
 
     @Test
     public void testNestedSimpleSelectUsesFetch() throws Exception {
         QueryThenFetch qtf = e.plan(
             "select x, i from (select x, i from t1 order by x asc limit 10) ti order by x desc limit 3");
-        Collect collect = (Collect) qtf.subPlan();
-        assertThat(collect.collectPhase().projections(), Matchers.contains(
+        List<Projection> projections = ((Collect) qtf.subPlan()).collectPhase().projections();
+        assertThat(projections, Matchers.contains(
             instanceOf(TopNProjection.class),
             instanceOf(TopNProjection.class),
             // TODO: We can optimize this to delay fetch until after the OrderedTopNProjection
             instanceOf(FetchProjection.class),
-            instanceOf(OrderedTopNProjection.class)
-        ));
+            instanceOf(OrderedTopNProjection.class)));
+        assertThat(projections.get(3).outputs()
```

Comment:
Yep you're right I can remove this assertion. The problem was only for join and we didn't have a test for that.